Python3 fails to build on Lion

`Brew install -vd Python3` fails, here is output
`clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic 
-DNDEBUG -g -fwrapv -O3 -Wall    -std=c99 -Wextra -Wno-unused-result -Wno-unused
-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-
function-declaration -fvisibility=hidden  -I./Include/internal  -I. -I./Include 
-I/usr/local/opt/tcl-tk/include -I/usr/local/opt/tcl-tk/include  -DPy_BUILD_CORE
_BUILTIN  -DPy_BUILD_CORE_BUILTIN -I./Include/internal -c ./Modules/signalmodule
.c -o Modules/signalmodule.o
fatal error: error in backend: Cannot select: 0x102565410: i8,ch = AtomicLoad 0x
101ae45a0, 0x102565110<Volatile LD1[@is_tripped.0.b]> [ID=11]
  0x102565110: i64 = X86ISD::WrapperRIP 0x102565810 [ID=10]
    0x102565810: i64 = TargetGlobalAddress<i1* @is_tripped.0.b> 0 [ID=8]
In function: _PyErr_CheckSignalsTstate
make: *** [Modules/signalmodule.o] Error 1
make: *** Waiting for unfinished jobs....
/usr/local/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: make 
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 1
==> Formula
Path: /usr/local/Library/Formula/python3.rb
==> Configuration
HOMEBREW_VERSION: 0.9.5
ORIGIN: (none)
HEAD: (none)
Last commit: never
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://ia904500.us.archive.org/24/items/tigerbrew
CPU: dual-core 64-bit penryn
OS X: 10.7.5-x86_64
Curl: /usr/local/Library/Homebrew/vendor/portable-curl/current/bin/curl
Xcode: 4.6.3
CLT: 4.6.0.0.1.1365549073
LLVM-GCC: build 2336
Clang: 4.2 build 425
X11: 2.6.5 => /opt/X11
System Ruby: 2.3.3-222 => /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3
/bin/ruby
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/1.8/us
r/bin/ruby
Java: 1.6.0_65-b14-462
==> ENV
HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
CPPFLAGS: -I/usr/local/opt/tcl-tk/include
LDFLAGS: -L/usr/local/opt/tcl-tk/lib
MAKEFLAGS: -j2
CMAKE_PREFIX_PATH: /usr/local/opt/readline:/usr/local/opt/sqlite:/usr/local/opt/
gettext:/usr/local/opt/perl:/usr/local/opt/openssl3:/usr/local/opt/bzip2:/usr/lo
cal/opt/zlib:/usr/local/opt/tcl-tk:/usr/local/opt/libffi:/usr/local
CMAKE_INCLUDE_PATH: /usr/include/libxml2:/System/Library/Frameworks/OpenGL.frame
work/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /System/Library/Frameworks/OpenGL.framework/Versions/Current
/Libraries
PKG_CONFIG_PATH: /usr/local/opt/xz/lib/pkgconfig:/usr/local/opt/readline/lib/pkg
config:/usr/local/opt/sqlite/lib/pkgconfig:/usr/local/opt/openssl3/lib/pkgconfig
:/usr/local/opt/bzip2/lib/pkgconfig:/usr/local/opt/zlib/lib/pkgconfig:/usr/local
/opt/tcl-tk/lib/pkgconfig:/usr/local/opt/libffi/lib/pkgconfig
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Library/ENV/pkgconfig/10.7
ACLOCAL_PATH: /usr/local/opt/gettext/share/aclocal:/usr/local/share/aclocal
PATH: /usr/local/Library/ENV/4.3:/usr/local/opt/xz/bin:/usr/local/opt/pkg-config
/bin:/usr/local/opt/sqlite/bin:/usr/local/opt/gdbm/bin:/usr/local/opt/gettext/bi
n:/usr/local/opt/perl/bin:/usr/local/opt/openssl3/bin:/usr/local/opt/bzip2/bin:/
usr/local/opt/tcl-tk/bin:/usr/bin:/bin:/usr/sbin:/sbin

Error: python3 3.10.16 did not build
Logs:
     /Users/Mylogin/Library/Logs/Homebrew/python3/01.configure
     /Users/Mylogin/Library/Logs/Homebrew/python3/01.configure.cc
     /Users/Mylogin/Library/Logs/Homebrew/python3/02.make
     /Users/Mylogin/Library/Logs/Homebrew/python3/02.make.cc
     /Users/Mylogin/Library/Logs/Homebrew/python3/config.log

/usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/net/http.r
b:933:in `connect_nonblock': Failed to connect to: https://api.github.com/search
/issues?q=python3+repo:mistydemeo/tigerbrew+in:title+state:open&per_page=100 (Gi
tHub::Error)
SSL_connect returned=1 errno=0 state=error: certificate verify failed
        from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3
.0/net/http.rb:933:in `connect'
        from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3
.0/net/http.rb:863:in `do_start'
        from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3
.0/net/http.rb:852:in `start'
        from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3
.0/open-uri.rb:319:in `open_http'
        from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3
.0/open-uri.rb:737:in `buffer_open'
        from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3
.0/open-uri.rb:212:in `block in open_loop'
        from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3
.0/open-uri.rb:210:in `catch'
        from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3
.0/open-uri.rb:210:in `open_loop'
        from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3
.0/open-uri.rb:151:in `open_uri'
        from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3
.0/open-uri.rb:717:in `open'
   from /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3
.0/open-uri.rb:31:in `open'
        from /usr/local/Library/Homebrew/utils.rb:435:in `open'
        from /usr/local/Library/Homebrew/utils.rb:465:in `issues_matching'
        from /usr/local/Library/Homebrew/utils.rb:497:in `issues_for_formula'
        from /usr/local/Library/Homebrew/exceptions.rb:209:in `fetch_issues'
        from /usr/local/Library/Homebrew/exceptions.rb:205:in `issues'
        from /usr/local/Library/Homebrew/exceptions.rb:248:in `dump'
        from /usr/local/Library/brew.rb:170:in `rescue in <main>'
        from /usr/local/Library/brew.rb:61:in `<main>'`

Any idea on how to solve this?

> > The path for clang is /usr/bin/clang, so I presume it came from Apple, probably XCode
> 
> Ok, I don't have XCode installed, I just use the command line tools bundle, but I'll take a look on my side.
> 
> Just so you can side step the issue temporarily, can you try `brew install python3 --cc=llvm-gcc` to see if that builds for you?'
`llvm-gcc -L/usr/local/opt/tcl-tk/lib -L/usr/local/opt/tcl-tk/lib   -Wl,-stack_si
ze,1000000  -framework CoreFoundation Python.framework/Versions/3.10/Python -o P
rograms/_testembed Programs/_testembed.o  -lintl -ldl   -framework CoreFoundatio
n    
DYLD_FRAMEWORK_PATH=/private/tmp/python320250223-89059-n26gx/Python-3.10.16 ./py
thon.exe -E -S -m sysconfig --generate-posix-vars ;\
        if test $? -ne 0 ; then \
                echo "generate-posix-vars failed" ; \
                rm -f ./pybuilddir.txt ; \
                exit 1 ; \
        fi
python.exe(99813) malloc: *** error for object 0xf0e0d0c0b0a0908: pointer being 
freed was not allocated
*** set a breakpoint in malloc_error_break to debug
/bin/sh: line 1: 99813 Abort trap: 6           DYLD_FRAMEWORK_PATH=/private/tmp/
python320250223-89059-n26gx/Python-3.10.16 ./python.exe -E -S -m sysconfig --gen
erate-posix-vars
generate-posix-vars failed
make: *** [pybuilddir.txt] Error 1

 llvm-gcc -v
Using built-in specs.
Target: i686-apple-darwin11
Configured with: /private/var/tmp/llvmgcc42/llvmgcc42-2336.11~182/src/configure 
--disable-checking --enable-werror --prefix=/Applications/Xcode.app/Contents/Dev
eloper/usr/llvm-gcc-4.2 --mandir=/share/man --enable-languages=c,objc,c++,obj-c+
+ --program-prefix=llvm- --program-transform-name=/^[cg][^.-]*$/s/$/-4.2/ --with
-slibdir=/usr/lib --build=i686-apple-darwin11 --enable-llvm=/private/var/tmp/llv
mgcc42/llvmgcc42-2336.11~182/dst-llvmCore/Developer/usr/local --program-prefix=i
686-apple-darwin11- --host=x86_64-apple-darwin11 --target=i686-apple-darwin11 --
with-gxx-include-dir=/usr/include/c++/4.2.1
Thread model: posix
gcc version 4.2.1 (Based on Apple Inc. build 5658) (LLVM build 2336.11.00)`

--- a/Library/Formula/python3.rb
+++ b/Library/Formula/python3.rb
@@ -48,6 +48,9 @@ class Python3 < Formula
   # Add Support for OS X before 10.6
   # from macports/lang/python310/files/patch-threadid-older-systems.diff
   # and macports/lang/python310/files/patch-no-copyfile-on-Tiger.diff
+  # Avoid Apple LLVM 4.2 for atomic operations
+  # https://bugs.python.org/issue24844
+  # macports/lang/python310/files/patch-configure-xcode4bug.diff
   patch :p0, :DATA
 
   # setuptools remembers the build flags python is built with and uses them to
@@ -496,4 +499,31 @@ __END__
 +@unittest.skipIf(not MACOS or not hasattr(posix, "_fcopyfile"), 'macOS with posix._fcopyfile only')
  class TestZeroCopyMACOS(_ZeroCopyFileTest, unittest.TestCase):
      PATCHPOINT = "posix._fcopyfile"
+ 
+--- configure.orig
++++ configure
+@@ -17359,6 +17359,24 @@
+     int main(void) {
+       __atomic_store_n(&val, 1, __ATOMIC_SEQ_CST);
+       (void)__atomic_load_n(&val, __ATOMIC_SEQ_CST);
++
++      /* https://bugs.python.org/issue24844 */
++      #define VERSION_CHECK(cc_major, cc_minor, req_major, req_minor) \
++          ((cc_major) > (req_major) || \
++          (cc_major) == (req_major) && (cc_minor) >= (req_minor))
++      #if defined(__clang__)
++          #if defined(__apple_build_version__)
++              // either one test or the other should work
++              // #if __apple_build_version__ < 5000000
++              #if !VERSION_CHECK(__clang_major__, __clang_minor__, 5, 0)
++                  #error
++              #endif
++          // not sure if this is 3.3 or 3.4
++          #elif !VERSION_CHECK(__clang_major__, __clang_minor__, 3, 3)
++              #error
++          #endif
++      #endif
++
+       return 0;
+     }
  

--- a/Library/Formula/python3.rb
+++ b/Library/Formula/python3.rb
@@ -77,8 +77,6 @@ class Python3 < Formula
       --with-system-ffi
     ]
 
-    args << "--without-gcc" if ENV.compiler == :clang
-
     cflags   = []
     ldflags  = []
     cppflags = []

